ViewEnvironmentUI/Sources/ViewEnvironmentPropagating.swift

```diff
@@ -176,6 +180,31 @@ extension ViewEnvironmentPropagating {
         }
     }
 
+    /// Adds a `ViewEnvironment` customization to this node.
+    ///
+    /// These customizations will occur before the node's `customize(environment:)` in cases where
+    /// this node conforms to `ViewEnvironmentObserving`, and will occur the order in which they
+    /// were added.
+    ///
+    /// The customization will only be active for as long as the returned lifetime is retained or
+    /// until `remove()` is called on it.
+    ///
+    @_spi(ViewEnvironmentWiring)
+    public func addEnvironmentCustomization(
```

Comment:
just to better my own understanding – should this trigger the 'setNeedsEnvironmentUpdate' (or whatever its name is) logic, or does that responsibility lie at a different level?

Reply:
Yeah—I'm fairly torn on whether or not this should automatically call `setNeedsEnvironmentUpdate()` on your behalf. You'll almost always want to call it after adding a customization, but I figured there may be cases where you'd prefer to defer that call or manually manage when the invalidation occurs (e.g. for performance reasons). `setNeedsEnvironmentUpdate()` does trigger `environmentDidChange()` calls which are not deferred to the next layout cycle, and we use this to perform updates in some cases within WorkflowUI and Market. For example, WorkflowHostingController will re-render its content whenever `environmentDidChange()` is called.

The current shape also makes it a bit easier to ensure events happen at more appropriate times. For example, if you were to be adding a customization to a new child VC in a container VC, you might have something like this:

```swift
addChild(viewController)
customizationLifetime = viewController
    .addEnvironmentCustomization(makeCustomization(for: column))
if let view = viewIfLoaded {
    view.add(viewController.view)
}
viewController.didMove(toParent: self)
viewController.setNeedsEnvironmentUpdate()
```
 This order of operations means:
* The customization and parent/child relationship is established before the child VC's view is (conditionally) loaded. This ensures any `environment` access during `viewDidLoad` would have the appropriate environment wiring path and customizations.
* `setNeedsEnvironmentUpdate()` is called after the parent/child relationship is fully established (`didMove(toParent:)` has been called). I'm not aware of any real issues that could come from calling this before `didMove(toParent:)`, but it seems like the most appropriate time in the VC lifecycle IMO.

I'm very open to automatically calling `setNeedsEnvironmentUpdate()` automatically when a customization is added if we prefer the safety here!